### PR TITLE
Master lps 190858 | Add tests on generate openAPI for api application definition

### PR DESCRIPTION
**Pull request by yanan**
**Background**
- Add functional tests

**Note**
- I've rebased to previous https://github.com/liferay-headless/liferay-portal/pull/1445 since the current test also use relatedEndpoint in ApplicationAPI.macro

**Test Plan**
- Given APIApplication ‘my-app’ with APIEndpoints with related APISchemas created
When I publish the APIApplication 
Then I can see ‘my-app’ on the REST Applications list of API Explorer
- When APIApplication ‘my-app’ of status ‘unpublished’ with APIEndpoints with related APISchemas created
Then I cannot see ‘my-app’ on the REST Applications list of API Explorer
- Given APIApplication ‘my-app’ of status ‘published’ with APIEndpoints with related APISchemas created
When I change state to ‘unpublish’
Then I cannot see ‘my-app’ on the REST Applications list of API Explorer
- Given APIApplication ‘my-app’ with APIEndpoints with related APISchemas created
And Given I publish the APIApplication 
When I click ‘my-app’ on the REST Applications list of API Explorer
Then I am redirected to the OpenApi page with endpoints and schemas details
And Then no “Failed to load API definition.” error

**Jira Ticket:**
- https://liferay.atlassian.net/browse/LPS-194106

### CODE REVIEW

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
ci:test:sf

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/GenerateOpenAPIFromAPIApplicationDefinition.testcase [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+		if (${testLiferayVirtualInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@priority = 5
+	test CanGenerateOpenApiFromApiApplicationDefinition {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
```
Comment (magjed4289) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/GenerateOpenAPIFromAPIApplicationDefinition.testcase:29]: Is there any specific reason we don't use the virtual instances for these tests? 🤔

Comment (yanan) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/GenerateOpenAPIFromAPIApplicationDefinition.testcase:29]: The reason I added `test.liferay.virtual.instance = "false"` is we haven't point out the tests are tested on a virtual instance in test plan ticket :smile: @magjed4289 Do we need to run all the tests on a virtual instance?

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
Failures are not related, and added tickets in Testray build.

**Comment by yanan** [comment]
ci:forward

**Pull request closed without merging**